src/grokcore/component/compat.py

```diff
@@ -0,0 +1,31 @@
+import sys
+import martian
+from zope.interface import classImplements
+from zope.interface.interfaces import IInterface
+
+if sys.version_info[0] < 3:
+    from zope.interface import implements
+else:
+
+    class implements(martian.Directive):
+        scope = martian.CLASS
+        store = martian.ONCE
+        default = None
+        
+        def factory(self, *interfaces):
+            for interface in interfaces:
+                if not IInterface.providedBy(interface):
+                    raise Exception('%s is not a interface' % repr(interface))
+            return interfaces
+
+
+    class ClassImplementerGrokker(martian.ClassGrokker):
```

Comment:
I'm unhappy about using different implementations for `implements()` for different Python versions.  I assume there are subtle semantic differences (e.g. one works right away on import, the other works only if you run a grokker on the source file).

I'd say use the grokker version for all Python versions.  (Or do what everybody else is doing and stop using implements() and use the @implementer class decoractor everywhere.)